### ISSUE TITLE
Fix Request: AvoidUnnecessaryStringBuilderCreation fails with two StringBuilders

### ISSUE THREAD

**Issue by sdalemans** [jborgers/PMD-jPinpoint-rules#106]
The AvoidUnnecessaryStringBuilderCreation rule causes problems when there are more StringBuilders within the same block, example code below results in:

> had problem: A sequence of more than one item is not allowed as the second argument of starts-with() ("one", "two")

```java
public class Foo {
    public void bar(Foo bar) {
        StringBuilder one = new StringBuilder();
        StringBuilder two = new StringBuilder();
        
        one.append("one");
        one.append("one-two");
        two.append("two");
        two.append("two-two");

        one.toString();
        two.toString();
    }
}
```
